proposals/lip-0062.md

```diff
@@ -25,6 +25,12 @@ Introducing memory efficient signing allows all objects used in the Lisk protoco
 
 With the current proposal, signatures are computed against the hash of the message to be signed and hence always computed for a byte string of small constant size.
 
+Lisk Elements currently contains another Ed25519 signature scheme [that uses double pre-hashing](https://github.com/LiskHQ/lisk-sdk/blob/0f1296c62585f26f3291f3e1a01aa757801f93f0/elements/lisk-cryptography/src/ed.ts#L99-L134). This scheme is only used for non-protocol related offchain signatures. More precisely the _sign message_ feature in Lisk Desktop is using it, and it is implemented in Lisk Commander without a concrete use case. The usage of double pre-hashing next to single pre-hashing yields a small risk for collisions. Notice that the existing double pre-hashing signature scheme prepends the tag `TAG = 0x15 + b"Lisk Signed Message:\n"` and the length of the message to the message prior to hashing. If one finds messages `m1` and `m2`, a message tag `MST` according to [LIP 0037](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0037.md#message-tags-1) and a chain ID `chainID` such that
+```
+sha256(TAG + len(m1) + m1) = MST + chainID + m2
```

Comment:
I would expand on the danger of the collision and explain that if `m2` is a transaction by someone, and `len(MST + chainID + m2) == 32` and an attacker finds a correct `m1` then they can pretend that they signed the message `TAG + len(m1) + m1`. I think this is the only reasonable attack scenario, or do I misunderstand something?

Reply:
I expanded a bit on it, but kept it more general and close to the attack scenario described in the audit report. The example with a transaction would also not work as the binary message would be too large.